src/tweenable.rs

```diff
@@ -561,72 +551,122 @@ impl<T> Tweenable<T> for Sequence<T> {
     }
 
     fn set_progress(&mut self, progress: f32) {
-        self.times_completed = if progress >= 1. { 1 } else { 0 };
-        let progress = progress.clamp(0., 1.); // not looping
-                                               // Set the total sequence progress
-        let total_elapsed_secs = self.duration().as_secs_f64() * progress as f64;
-        self.time = Duration::from_secs_f64(total_elapsed_secs);
-
-        // Find which tween is active in the sequence
-        let mut accum_duration = 0.;
-        for index in 0..self.tweens.len() {
-            let tween = &mut self.tweens[index];
-            let tween_duration = tween.duration().as_secs_f64();
-            if total_elapsed_secs < accum_duration + tween_duration {
-                self.index = index;
-                let local_duration = total_elapsed_secs - accum_duration;
-                tween.set_progress((local_duration / tween_duration) as f32);
-                // TODO?? set progress of other tweens after that one to 0. ??
-                return;
+        // Optimize the boundary conditions
+        if progress < 1e-5 {
+            self.rewind();
+            return;
+        } else if progress > 1. - 1e-5 {
+            self.elapsed = self.duration;
+            self.index = self.tweens.len();
+            return;
+        }
+
+        self.elapsed = self.duration.mul_f32(progress.clamp(0., 1.));
```

Comment:
The interface for `set_progress()` should accept values > `1.`, which allows jumping to iteration N of a repeating animation. Looping sequences are not yet implemented but there's work on-going for them (#16).